storage/src/block_storage.rs

```diff
@@ -446,10 +444,19 @@ impl BlockPrimaryIndex {
     }
 
     #[inline]
-    fn iterator(&self) -> Result<IteratorWithSchema<Self>, StorageError> {
-        self.kv
-            .iterator(IteratorMode::Start)
-            .map_err(StorageError::from)
+    fn iterator(&self) -> Result<Vec<BlockHash>, StorageError> {
+        let items = self
+            .kv
+            .find(IteratorMode::Start, None, Box::new(|(_, _)| Ok(true)))?;
+        let mut results = vec![];
```

Comment:
@mambisi @jurajselep @tizoc @akoptelov @sebastiencs 
I guess this is too expensive,
everytime I call `iterator()` it will load the whole database?

and another thing is that to add to empty `vec![]` wihtout capacity, every push will trigger lots of reallocations

@mambisi 
does `self.kv.` have any function for `len()` or something like that?

Reply:
calling `len()` on sled is expensive
> From the sled documentation
```
Returns the number of elements in this tree.

Beware: performs a full O(n) scan under the hood.
```